pkg/distributor/distributor.go

```diff
@@ -485,11 +485,30 @@ func (p *pushTracker) doneWithResult(err error) {
 	}
 }
 
+func (d *Distributor) waitSimulatedLatency(ctx context.Context, tenantID string, start time.Time) {
+	latency := d.validator.Limits.SimulatedPushLatency(tenantID)
+	if latency > 0 {
```

Comment:
I believe this line can be removed as the condition is tested again when we check if `wait > 0`. For example, when `latency` is 0, then `wait` is negative.

However, I suppose someone could set `latency` to a negative number though? Is that a concern?